kimchi/src/verifier_index.rs

```diff
@@ -183,6 +190,9 @@ where
                 cell
             },
 
+            public_input_size: self.cs.public,
+            recursion_challenges: todo!(),
```

Comment:
Take it as an input to the constraint system builder, pass it through from there.